src/bacnet/basic/service/s_cov.c

```diff
@@ -174,3 +174,67 @@ uint8_t Send_COV_Subscribe(
 
     return invoke_id;
 }
+
+/** Sends a COV Subscription request directly to an address.
+ * @ingroup DSCOV
+ *
+ * @param dest [in] address of destination device
+ * @param max_apdu [in] max size of apdu
+ * @param cov_data [in]  The COV subscription information to be encoded.
+ * @return invoke id of outgoing message, or 0 if communication is disabled or
+ *         no slot is available from the tsm for sending.
+ */
+uint8_t Send_COV_Subscribe_Direct(
+    BACNET_ADDRESS *dest, unsigned max_apdu, BACNET_SUBSCRIBE_COV_DATA *cov_data)
+{
+    BACNET_ADDRESS my_address;
+    uint8_t invoke_id = 0;
+    int len = 0;
+    int pdu_len = 0;
+    int bytes_sent = 0;
+    BACNET_NPDU_DATA npdu_data;
+
+    if (!dcc_communication_enabled()) {
+        return 0;
+    }
+    /* is there a tsm available? */
+    invoke_id = tsm_next_free_invokeID();
+    if (invoke_id) {
```

Comment:
`if (!invoke_id) return invoke_id;` would remove one level of indentation and would make this more readable